File: 101/rework_exercises/easy_3/palindromic_strings.rb
# palindromic_strings.rb

# Write a method that returns true if the string passed as an argument
# is a palindrome, false otherwise. A palindrome reads the same
# forward and backward. For this exercise, case matters as does
# punctuation and spaces.

# Understanding:
# Palindrome is the same forward and backwards
# Return a boolean
# Case sensitive
#   can just reverse

def palindrome?(str)
  backwards_string = str.reverse
  str == backwards_string
end

# Examples:

puts palindrome?('madam') == true
puts palindrome?('Madam') == false          # (case matters)
puts palindrome?("madam i'm adam") == false # (all characters matter)
puts palindrome?('356653') == true

File: 210/js_small_problems/easy_3/palindromic_strings.js
// palindromic_strings.js

function isPalindrome(str) {
  var reversedString = str.split('').reverse().join('');
  if (str === reversedString) { return true}
  return false;
}

console.log(isPalindrome('madam'));               // true
console.log(isPalindrome('Madam'));               // false (case matters)
console.log(isPalindrome("madam i'm adam"));      // false (all characters matter)
console.log(isPalindrome('356653'));              // true